var/spack/repos/builtin/packages/steps/package.py

```diff
@@ -12,51 +14,127 @@ class Steps(CMakePackage):
+    depends_on("random123", when="~bundle")
+    depends_on("sundials@:2.99.99+int64", when="~bundle")
 
     patch("for_aarch64.patch", when="target=aarch64:")
 
+    def patch(self):
+        # easylogging requires compilation by
+        # its dependents: splice in disabling all errors
+        filter_file(r"(-Wno-double-promotion)", r"-Wno-error \1", "src/steps/util/CMakeLists.txt")
+
     def cmake_args(self):
-        args = []
-        spec = self.spec
-
-        if "+native" in spec:
-            args.append("-DTARGET_NATIVE_ARCH:BOOL=True")
-        else:
-            args.append("-DTARGET_NATIVE_ARCH:BOOL=False")
-
-        if "+lapack" in spec:
-            args.append("-DUSE_BDSYSTEM_LAPACK:BOOL=True")
-        else:
-            args.append("-DUSE_BDSYSTEM_LAPACK:BOOL=False")
-
-        if "+petsc" in spec:
-            args.append("-DUSE_PETSC:BOOL=True")
-        else:
-            args.append("-DUSE_PETSC:BOOL=False")
-
-        if "+mpi" in spec:
-            args.append("-DUSE_MPI:BOOL=True")
-        else:
-            args.append("-DUSE_MPI:BOOL=False")
-
-        args.append("-DBLAS_LIBRARIES=" + spec["blas"].libs.joined(";"))
+        python_interpreter = self.spec["python"].prefix.bin.python + str(
+            self.spec["python"].version.up_to(1)
+        )
+        args = [
+            self.define("STEPS_INSTALL_PYTHON_DEPS", False),
+            self.define_from_variant("BUILD_STOCHASTIC_TESTS", "stochtests"),
+            self.define_from_variant("BUILD_TESTING", "codechecks"),
+            self.define_from_variant("ENABLE_CODECOVERAGE", "coverage"),
+            self.define_from_variant("STEPS_ENABLE_ERROR_ON_WARNING", "codechecks"),
+            self.define_from_variant("STEPS_TEST_FORMATTING", "codechecks"),
+            self.define_from_variant("STEPS_USE_CALIPER_PROFILING", "caliper"),
+            self.define_from_variant("STEPS_USE_DIST_MESH", "distmesh"),
+            self.define_from_variant("STEPS_USE_LIKWID_PROFILING", "likwid"),
+            self.define_from_variant("TARGET_NATIVE_ARCH", "native"),
+            self.define_from_variant("USE_BDSYSTEM_LAPACK", "lapack"),
+            self.define_from_variant("USE_MPI", "mpi"),
+            self.define_from_variant("USE_PETSC", "petsc"),
+            "-DBLAS_LIBRARIES=" + self.spec["blas"].libs.joined(";"),
+            f"-DPYTHON_EXECUTABLE={python_interpreter}",
+        ]
+
+        args.extend(
+            [
+                self.define_from_variant(f"USE_BUNDLE_{bundle}", "bundle")
+                for bundle in ["EASYLOGGINGPP", "OMEGA_H", "RANDOM123", "SUNDIALS", "SUPERLU_DIST"]
+            ]
+        )
+
         return args
 
+    @property
+    def build_targets(self):
+        targets = []
+        if "+coverage" in self.spec:
+            if self.compiler.name != "gcc":
+                raise ValueError(
+                    "Package " + self.name + " build with coverage enabled requires GCC to build"
+                )
```

Comment:
FYI, this can be added as a directive, once #36286 is merged. The directive would be:
```python
requires("%gcc", when="+coverage", msg="GCC is required to build with coverage")
```

Reply:
Very nice improvement indeed. Should we wait for #36286 to be merged?

Reply:
Not really, but you can keep an eye on PRs, and submit an improvement to this recipe once that happens!